### ISSUE TITLE
Suddenly "dotnet watch" runs in "Production" environment

### ISSUE THREAD

**Issue by Apollo3zehn** [dotnet/aspnetcore#44363]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Yesterday I first read about the new Ubuntu packages (https://devblogs.microsoft.com/dotnet/dotnet-6-is-now-in-ubuntu-2204/), today morning I updated my system (Kubuntu 22.04) and then I suddenly had the problem of missing SDKs. I thought it was maybe related to the new Ubuntu packages and conflicting versions. So I removed the old packages and cleanly reinstalled the SDK from the Ubuntu repository. I was also able to install the wasm-tools workload but now "dotnet watch" runs in "Production" environment and breaks my app. I am not 100% sure it is due to the Ubuntu provided SDK but it is very likely.

I tried to revert to the Microsoft package source to get the old behavior but I failed to find proper instructions to do so. Then I tried the Snap version and that fails with a segmentation fault ...

The only (maybe) related issue I found is this one (https://github.com/dotnet/core/issues/7715)

As a workaround I extended my tasks.json with a new environment variable:

```json
{
    "version": "2.0.0",
    "options": {
        "env": {
            "ASPNETCORE_ENVIRONMENT": "Development"
        }
    },
    "tasks": [
        ...
```

### Expected Behavior

I expect that `dotnet watch` runs in the "Development" environment.

### Steps To Reproduce

- Run `dotnet new blazorwasm --hosted`. 
- Open the project in vscode
- Add `Console.WriteLine("The current environment is " + Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT"));` to `Program.cs` in the `Server` project
- Run "watch" task (Ctrl+Shift+P -> "Run task" -> "watch")

The output is `The current environment is <empty>`

and a few lines later:

```
info: Microsoft.Hosting.Lifetime[0]
      Hosting environment: Production
```

### .NET Version

6.0.108

### Anything else?

`dotnet --info`:

```
.NET SDK (reflecting any global.json):
 Version:   6.0.108
 Commit:    4e3a463d2b

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  22.04
 OS Platform: Linux
 RID:         ubuntu.22.04-x64
 Base Path:   /usr/lib/dotnet/dotnet6-6.0.108/sdk/6.0.108/

global.json file:
  Not found

Host:
  Version:      6.0.8
  Architecture: x64
  Commit:       55fb7ef977

.NET SDKs installed:
  6.0.108 [/usr/lib/dotnet/dotnet6-6.0.108/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.8 [/usr/lib/dotnet/dotnet6-6.0.108/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.8 [/usr/lib/dotnet/dotnet6-6.0.108/shared/Microsoft.NETCore.App]

Download .NET:
  https://aka.ms/dotnet-download

Learn about .NET Runtimes and SDKs:
  https://aka.ms/dotnet/runtimes-sdk-info

```

**Comment by jswolf19** [comment]
I'm having a similar problem. It seems that `dotnet watch run` is using the first profile where `commandName` is "Project" in launchSettings.json even if a different launch profile is specified on the command line.
For example, if I have profiles "P1" and "P2" in my launchSettings defined in that order, then the following command will use "P1":

```
dotnet watch run <csproj/sln> --launch-profile P2
```

It seems that I can work around this by passing launch profile to dotnet-run

```
dotnet watch run <csproj/sln> -- --launch-profile P2
```

**Comment by r-work** [comment]
Just ran into this issue, as commented above, it seems like this starts happening if there are any comments in `launchSettings.json`.
As soon as I removed the comments, it started working again.

**Comment by mkArtakMSFT** [comment]
Thanks for reporting this, everyone.
Can somebody try out the latest 8.0 release and see if this issue is resolved there (we think it is).

**Comment by tk-at-work** [comment]
Just adding this for anyone else having a similar issue. We edited our launchSettings.json a few weeks ago together with a lot of other changes. Later we found that `dotnet watch run` did crash the project because the secret manager was not injecting secrets.

Just a few minutes ago i found out, that the app was running with Production AspEnvironment. Then i found this, as some hinted at the launchSettings:

![image](https://github.com/dotnet/aspnetcore/assets/117629410/dc4962ec-8d49-4be3-b023-89cdba229572)

As the comma seems to be fine for JSON in general, `dotnet watch run` does not like it and discards the launch profile, leading to running in Production mode. We did not had any comments in this file, but removing the trailing comma solved the issue.
